packages/common/src/allo/transaction-sender.ts

```diff
@@ -185,24 +200,46 @@ export function createMockTransactionSender(): TransactionSender & {
       return {
         transactionHash: txHash,
         blockHash: `0x${Math.random().toString(16).slice(2)}` as Hex,
-        blockNumber: 1n,
+        blockNumber: BigInt(1),
         logs: [],
       };
     },
+
+    async address(): Promise<Address> {
+      return ("0x" + "0".repeat(40)) as Address;
+    },
   };
 }
 
+export async function sendRawTransaction(
+  sender: TransactionSender,
+  args: TransactionData
+): Promise<Result<Hex>> {
+  try {
+    const tx = await sender.send({
+      to: args.to,
+      data: args.data,
+      value: args.value,
+    });
+    return success(tx);
+  } catch (err) {
+    return error(
+      new AlloError(`Failed to send raw transaction: ${String(err)}`, err)
+    );
+  }
+}
+
 export async function sendTransaction(
   sender: TransactionSender,
-  args: Parameters<typeof encodeFunctionData>[0] & { address: Address }
+  args: EncodeFunctionDataParameters<Abi, string> & { address: Address }
```

Comment:
This loses the type safety, you should be able to write the signature like this to preserve it:

```typescript
export async function sendTransaction<
  TAbi extends Abi,
  TFunctionName extends string,
>(
  sender: TransactionSender,
  args: EncodeFunctionDataParameters<TAbi, TFunctionName> & { address: Address }
): Promise<Result<Hex>> {
```

This way function names and their parameters will be typed and mistakes will be caught at compile time.

Once you make this change you will see that the  `updateProjectMetadata` call is wrong in the V1 implementation, we're passing a Hex value when we should be passing a `uint256`. If the `updateProjectMetadata` can receive both a Hex and Bigint to suppport both V1 and V2 we can make the interface type be something like `bigint | Hex` and force the implementatin to handle the union, instead of throwing away the types.

Reply:
Just noticed we're encoding the bigint before calling the `Allo.updateProjectMetadata` then I would say that instead of losing the types here we can just keep passing a Hex but decode it back to bigint before calling `sendTransaction`, I know it seems wasteful but it's a tradeoff.